src/data/finset/pointwise.lean

```diff
@@ -430,14 +349,98 @@ coe_injective.div_inv_monoid _ coe_one coe_mul coe_inv coe_div coe_pow coe_zpow
 protected def div_inv_monoid' [group_with_zero α] : div_inv_monoid (finset α) :=
 coe_injective.div_inv_monoid _ coe_one coe_mul coe_inv coe_div coe_pow coe_zpow'
 
-localized "attribute [instance] finset.mul_one_class finset.add_zero_class finset.semigroup
-  finset.add_semigroup finset.monoid finset.add_monoid finset.comm_monoid finset.add_comm_monoid
-  finset.div_inv_monoid finset.sub_neg_add_monoid finset.div_inv_monoid'"
-  in pointwise
+localized "attribute [instance] finset.comm_monoid finset.add_comm_monoid finset.div_inv_monoid
+  finset.sub_neg_add_monoid finset.div_inv_monoid'" in pointwise
 
 end instances
 
-/-! ### Finset addition/multiplication -/
+section mul_zero_class
+variables [decidable_eq α] [mul_zero_class α] {s t : finset α}
+
+/-! Note that `set` is not a `mul_zero_class` because `0 * ∅ ≠ 0`. -/
+
+lemma mul_zero_subset (s : finset α) : s * 0 ⊆ 0 := by simp [subset_iff, mem_mul]
+lemma zero_mul_subset (s : finset α) : 0 * s ⊆ 0 := by simp [subset_iff, mem_mul]
+
+lemma nonempty.mul_zero (hs : s.nonempty) : s * 0 = 0 :=
+s.mul_zero_subset.antisymm $ by simpa [mem_mul] using hs
+
+lemma nonempty.zero_mul (hs : s.nonempty) : 0 * s = 0 :=
+s.zero_mul_subset.antisymm $ by simpa [mem_mul] using hs
+
+end mul_zero_class
```

Comment:
I think I'd probably expect to see these before `div_inv_monoid`, perhaps just after `mul_one_class`

Reply:
Fair enough